modules/webbrowser/src/processors/webbrowserbase.cpp

```diff
@@ -93,44 +92,53 @@ void WebBrowserBase::executeJavaScript(const std::string& javascript, int startL
 
 void WebBrowserBase::reload() {
     isLoading_ = true;
-    browser_->GetMainFrame()->LoadURL(std::string{getSource()});
+    browser_->GetMainFrame()->LoadURL(getURL());
 }
 
 bool WebBrowserBase::isLoading() const { return isLoading_; }
 
 void WebBrowserBase::load(const std::filesystem::path& filename) {
     if (!filename.empty()) {
-        source_ = fmt::format("file://{}", filename.string());
+        url_ = fmt::format("file://{}", filename.string());
         reload();
-    } else if (!source_.empty()) {
-        source_.clear();
+    } else if (!url_.empty()) {
+        url_.clear();
         reload();
     }
 }
 
-void WebBrowserBase::load(std::string_view uri) {
-    if (source_ != uri) {
-        source_ = uri;
+void WebBrowserBase::load(std::string_view url) {
+    if (url_ != url) {
+        url_ = url;
         reload();
     }
 }
 
+void WebBrowserBase::clear() {
+    url_ = fmt::format(
+        "file://{}",
+        (util::getModulePath("WebBrowser", ModulePath::Data) / "html/empty.html").string());
+    reload();
+}
+
 void WebBrowserBase::setZoom(double zoom) {
     browser_->GetHost()->SetZoomLevel(cefutil::percentageToZoomLevel(zoom));
 }
 
 std::shared_ptr<BaseCallBack> WebBrowserBase::addLoadingDoneCallback(std::function<void()> f) {
-    return loadingDone_.addLambdaCallbackRaii(f);
+    return loadingDone_.add(std::move(f));
 }
 
-std::string_view WebBrowserBase::getSource() const {
+const std::string& WebBrowserBase::getURL() const {
 #ifndef NDEBUG
     // CEF does not allow empty urls in debug mode
-    if (source_.empty()) {
-        return "https://www.inviwo.org";
+    if (url_.empty()) {
```

Comment:
can't return this temp as a reference? better handle this in reload. or check in the load functions instead to set the url_